### ISSUE TITLE
GPU tracer in sagemaker TensorFlow

### ISSUE THREAD

**Issue by khu834** [aws/sagemaker-python-sdk#356]
### System Information
- **Framework (e.g. TensorFlow) / Algorithm (e.g. KMeans)**: TensorFlow
- **Framework Version**: 1.8
- **Python Version**: python 2.7
- **CPU or GPU**: GPU
- **Python SDK Version**: 
- **Are you using a custom image**: No

### Describe the problem
It would be really helpful to be able to run the CUPTI GPU tracer for tensorflow during training in SageMaker in order to get runtime statistics. Something like this: 

`run_metadata = tf.RunMetadata()`
`_, l, lr, predictions = sess.run(
            [optimizer, loss, learning_rate, train_prediction],
            feed_dict=feed_dict,
            options=tf.RunOptions(trace_level=tf.RunOptions.FULL_TRACE),
            run_metadata=run_metadata)`

Is there a way to pass an option into the container session.run(), such that we can see the run states in TensorBoard?

**Comment by ChoiByungWook** [comment]
I am not a experienced machine learning or TensorFlow expert, so anyone reading this please feel free to correct me or participate in the conversation.

But as of now, the SageMaker TensorFlow containers utilize Estimators to handle training. I assume CUPTI GPU tracer is in reference to the tf.RunMetadata(). Unfortunately it doesn't look like we can directly pass in the run_metadata to the estimator based on the following links.
- https://github.com/tensorflow/tensorflow/issues/13594
- https://stackoverflow.com/questions/45719176/how-to-display-runtime-statistics-in-tensorboard-using-estimator-api-in-a-distri?rq=1

However, based on the stackoverflow link, you might be able to utilize [tf.train.ProfilerHook](https://www.tensorflow.org/api_docs/python/tf/train/ProfilerHook). I believe this can be specified in your EstimatorSpec [estimator_fn](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/tensorflow/README.rst#preparing-the-tensorflow-training-script).

I apologize that this might not be the experience you were expecting.

**Comment by laurenyu** [comment]
Closing due to inactivity. Feel free to reopen if necessary.
